weave/trace_server/clickhouse_trace_server_batched.py

```diff
@@ -302,6 +322,53 @@ def calls_query_stream(
                 _ch_call_dict_to_call_schema_dict(ch_dict)
             )
 
+    def calls_delete(self, req: tsi.CallsDeleteReq) -> tsi.CallsDeleteRes:
+        if len(req.call_ids) > MAX_DELETE_CALLS_COUNT:
+            raise RequestTooLarge(
+                f"Cannot delete more than {MAX_DELETE_CALLS_COUNT} calls at once"
+            )
+
+        proj_cond = "project_id = {project_id: String}"
+        proj_params = {"project_id": req.project_id}
+
+        # get all parents
+        parents = self._select_calls_query(
+            req.project_id,
+            conditions=[proj_cond, "id IN {ids: Array(String)}"],
+            parameters=proj_params | {"ids": req.call_ids},
+        )
+
+        # get all calls with trace_ids matching parents
+        all_calls = self._select_calls_query(
+            req.project_id,
+            conditions=[proj_cond, "trace_id IN {trace_ids: Array(String)}"],
+            parameters=proj_params | {"trace_ids": [p.trace_id for p in parents]},
+        )
+
+        all_descendants = find_call_descendants(
+            root_ids=req.call_ids,
+            all_calls=all_calls,
+        )
+
+        deleted_at = datetime.datetime.now()
+        insertables = [
+            CallDeleteCHInsertable(
+                project_id=req.project_id,
+                id=call_id,
+                wb_user_id=req.wb_user_id,
+                deleted_at=deleted_at,
+            )
+            for call_id in all_descendants
+        ]
+        print(
+            f"Deleting {len(insertables)} calls:\n   ",
+            "\n    ".join([i.id for i in insertables]),
+        )
```

Comment:
how do we feel about print statements like this? There are many, and personally I think it looks fine: 

<img width="524" alt="Screenshot 2024-05-21 at 1 03 58 PM" src="https://github.com/wandb/weave/assets/19414170/c10d28f6-13fa-4754-ac44-6417c795f25d">